### ISSUE TITLE
`PickList`s inside `Component`s have missaligned popup menus

### ISSUE THREAD

**Issue by DoomDuck** [iced-rs/iced#2177]
### Is there an existing issue for this?

- [X] I have searched the existing issues.

### Is this issue related to iced?

- [X] My hardware is compatible and my graphics drivers are up-to-date.

### What happened?

When creating a `PickList` inside a `Component` their popup menu is misaligned when the widget is not in the top left corner.

![image](https://github.com/iced-rs/iced/assets/37777252/36fdb667-71e1-4943-b938-50d09c06efec)

I created a small repository to reproduce the error:
https://github.com/DoomDuck/iced_picklist_in_component

### What is the expected behavior?

I expect the popup to be aligned with the button.

### Version

crates.io release

### Operating System

Linux

### Do you have any log output?

```shell
No
```

**Comment by william-shere** [comment]
I believe this is fixed in 25006b9

You can pull the fixed version of iced by replacing the dependency in your `Cargo.toml` with:

```
iced = { git = "https://github.com/iced-rs/iced.git", rev = "25006b9c6f2ae909d86871d3a13631d518c07158", features = ["lazy"] }
```

or to pull the latest version:

```
iced = { git = "https://github.com/iced-rs/iced.git", rev = "f5c6fa7338713a0430dcdfc5bd53f72103144f22", features = ["lazy"] }
```
and then running
```
cargo update
```

**Comment by DoomDuck** [comment]
Thank you ! This has indeed been fixed !
The suggested tags do fix the problem thank you :+1:
